File: redsheet-portal/src/app/database-models/audit-request.ts
import { required, maxLength, range, nested } from '@rx/annotations';
import { AuditRecord,  } from './'
export class AuditRequest {
    constructor(auditRequest?: AuditRequest )  {
        let properties = [ "applicationModuleId", "applicationTimeZoneId", "auditRequestId", "createdDate", "mainRecordId", "requestMethod", "uri", "userId", "auditRecords",];
        for (let property of properties)
            if (auditRequest && auditRequest[property])
                this[property] = auditRequest[property];
    }
 
    @range(1,2147483647)
	applicationModuleId : number =   undefined;
 
    @range(1,2147483647)
	applicationTimeZoneId : number =   undefined;
 
	auditRequestId : number =   0 ;
 
    @required()
	createdDate : Date =   undefined;
 
    @range(1,2147483647)
	mainRecordId : number =   undefined;
 
    @required()
    @maxLength(20)
	requestMethod : string =   undefined;
 
    @required()
	uri : string =   undefined;
 
    @range(1,2147483647)
	userId : number =   undefined;
	@nested(AuditRecord)
	auditRecords: AuditRecord[];



}

File: RedSheet.DbEntities/Models/AuditRequest.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("AuditRequests", Schema = "dbo")]
	[Serializable]
	public partial class AuditRequest  
	{
 
	[Range(1, int.MaxValue)]
	public int ApplicationModuleId { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int ApplicationTimeZoneId { get; set; } 
 
	[DatabaseGenerated(DatabaseGeneratedOption.Identity)] 
	[Key]
	public int AuditRequestId { get; set; } 
 
	[Required]
	public System.DateTime CreatedDate { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int MainRecordId { get; set; } 
 
	[Required] 
	[MaxLength(20)]
	public string RequestMethod { get; set; } 
 
	[Required]
	public string Uri { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int UserId { get; set; } 
	[InverseProperty("AuditRequest")]
	public ICollection<AuditRecord> AuditRecords { get; set; } 


	public AuditRequest()
	{
	this.AuditRecords = new HashSet<AuditRecord>();

    }

    }
}

